File: Bug_Hunt_Practice/Question 3/solution.py
def is_palindrome(s):
    length = len(s)
    for i in range(length // 2):
        if s[i] != s[length - i - 1]:
            return False
    return True

# --- Main execution ---
text = "racecar"
result = is_palindrome(text)
print(f"Checking 'racecar'...")
print(f"Expected Output: True")
print(f"Actual Output:   {result}")

text2 = "hello"
result2 = is_palindrome(text2)
print(f"\nChecking 'hello'...")
print(f"Expected Output: False")
print(f"Actual Output:   {result2}")

File: Bug_Hunt_Practice/Question 3/solution.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindrome(std::string str) {
    int len = str.length();
    for (int i = 0; i < len / 2; i++) {
        if (str[i] != str[len - i]) {
            return false;
        }
    }
    return true;
}

int main() {
    string text = "racecar";
    bool result = isPalindrome(text);
    cout << "Checking 'racecar'..." << endl;
    cout << "Expected Output: true" << endl;
    cout << "Actual Output:   " << boolalpha << result << endl;
    
    string text2 = "hello";
    bool result2 = isPalindrome(text2);
    cout << "\nChecking 'hello'..." << endl;
    cout << "Expected Output: false" << endl;
    cout << "Actual Output:   " << boolalpha << result2 << endl;
    return 0;
}